terrautils/betydb.py

```diff
@@ -47,6 +48,128 @@ def get_bety_key():
                        "or create $HOME/.betykey.")
 
 
+def get_brapi_url(path=''):
+    """return brapi url from environment with optional path
+
+    Of 3 options string join, os.path.join and urlparse.urljoin, os.path.join
+    is the best at handling excessive / characters.
+    """
+
+    url = os.environ.get('BRAPI_URL', BRAPI_URL)
+    return os.path.join(url, path)
+
+
+def get_brapi_api(endpoint=None):
+    """return brapi API based on brapi url"""
+
+    url = get_brapi_url(path='v1/{}'.format(endpoint))
+    return url
+
+
+def get_brapi_study(studyDbId):
+    """return study from brapi based on brapi url"""
+    endpoint = 'studies'
+    study_url = get_brapi_api(endpoint)
+    request_params = {'studyDbId':studyDbId}
+    r = requests.get(url=study_url, params=request_params)
+    return r.json()
+
+def get_brapi_observationunits(studyDbId, pageSize=None):
+    endpoint = 'observationunits'
+    study_url = get_brapi_api(endpoint)
+    request_params = {'studyDbId': studyDbId}
+    if pageSize:
+        request_params['pageSize'] = pageSize
+    r = requests.get(url=study_url, params=request_params)
+    return r.json()
+
+def get_brapi_all_observation_units(studyDbId):
+    first_page_results = get_brapi_observationunits(studyDbId)
+    results_count = first_page_results['metadata']['pagination']['totalCount']
+    all_observation_units = get_brapi_observationunits(studyDbId, pageSize=results_count)
```

Comment:
don't ask for all results, just pagenate and append, you can probably make a function called `get_brapi_result(path, params)` which returns a list with all the data.

Reply:
this is moved to the brapi_get which now handles all pagination